### PR TITLE
feat: set seconds & milliseconds to zero

### PR DESCRIPTION
**Pull request by cseas**
## Description

[Jira][1]

Set both seconds and milliseconds to zero before making the GraphQL call for all queries. 
This is a perf improvement requested by backend.

### Testing

Tested on [test environment][2]

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

[1]: https://razorpay.atlassian.net/browse/OBSV-1145
[2]: https://hypertrace-test.concierge.stage.razorpay.in/services?time=5m

### CODE REVIEW

File: projects/common/src/time/relative-time-range.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ export class RelativeTimeRange implements TimeRange {
   public constructor(public readonly duration: Readonly<TimeDuration>) {
     this.endTime.setTime(Date.now());
     this.startTime.setTime(this.endTime.getTime() - this.duration.toMillis());
+
+    this.endTime.setSeconds(0, 0);
+    this.startTime.setSeconds(0, 0);
```
Comment (jaywalker21) [inline projects/common/src/time/relative-time-range.ts]: If it's fine, can we maintain this under a boolean and have it true by default for now. The reason being that, this is something very explicit and will help a lot when it comes to understanding the code in the future.
Also, maybe mentioning a comment would help in the future for anyone to know the context too.

I feel this will also affect other places apart from the top bar where time related stuff is used, but this shouldn't be an issue I guess, do test it thoroughly if possible though.

File: projects/components/src/time-range/time-range.component.test.ts [hunk 2]
```diff
@@ -150,4 +150,17 @@ describe('Time range component', () => {
     expect(refreshButton.label).toBe('Refresh - updated 6m ago');
     discardPeriodicTasks();
   }));
+
+  test('should set seconds and milliseconds to zero', () => {
```
Comment (jaywalker21) [inline projects/components/src/time-range/time-range.component.test.ts:154]: Can we also add a test case for relative time range too? That will ensure that both `sec` and `ms` are zero.

File: projects/components/src/time-range/time-range.component.ts [hunk 3]
```diff
@@ -108,6 +108,8 @@ export class TimeRangeComponent {
   }
 
   public setToFixedTimeRange(timeRange: FixedTimeRange): void {
+    timeRange.startTime.setSeconds(0, 0);
```
Comment (jaywalker21) [inline projects/components/src/time-range/time-range.component.ts:115]: Let's add a comment here for future reference, in case someone apart from us both reviews this code.

**Review by jaywalker21 — COMMENTED** [review]
Code looks good, couple of points to add comments for better understanding.

File: projects/common/src/time/relative-time-range.ts (hunk 1 above)
Comment (cseas) [reply projects/common/src/time/relative-time-range.ts]: ✅ Added a boolean with a default value of `true`

File: projects/components/src/time-range/time-range.component.test.ts (hunk 2 above)
Comment (cseas) [reply projects/components/src/time-range/time-range.component.test.ts:154]: ✅ Added test for relative time range

File: projects/components/src/time-range/time-range.component.ts (hunk 3 above)
Comment (cseas) [reply projects/components/src/time-range/time-range.component.ts:115]: ✅ Added comment

**Review by jaywalker21 — APPROVED** [review]
